server/src/test/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapperTests.java

```diff
@@ -176,6 +176,20 @@ public void testNullField() throws Exception {
         assertEquals(0, fields.length);
     }
 
+    public void testBlankFieldName() throws Exception {
+        DocumentMapper mapper = createDocumentMapper(fieldMapping(this::minimalMapping));
+        ParsedDocument parsedDoc = mapper.parse(source(b -> b.field("", "value")));
```

Comment:
you are mapping a field of type flattened:

```
{
  "properties" : {
    "field" : {
        "type" : "flattened"
    }
  }
}
```

and then submitting a doc like the following:

```
{
  "" : "value"
}
```

The empty field name does not belong to the flattened field, hence I would hope that your fix does not affect this specific field. On the other hand, if your doc has a field called `field`, with a sub-field with an empty name, your fix should make that scenario work while it previously didn't.

Reply:
Done.